lib/Client.php

```diff
@@ -117,6 +117,12 @@ public function read($size = null) {
         }
 
         $state = $this->state;
+        if (isset($state->buffer[0]) && ($size === null || isset($state->buffer[$size-1]))) {
```

Comment:
this should probably read:
```php
if (isset($state->buffer[$size === null ? 0 : $size - 1]))
```
There's no reason to check for both [0] and [$size - 1] if $size !== null